ledger/complete/wal/checkpoint_v6_test.go

```diff
@@ -158,6 +175,12 @@ func createMultipleRandomTriesMini(t *testing.T) []*trie.MTrie {
 	require.NoError(t, err, "update registers")
 	tries = append(tries, activeTrie)
 
+	// trie must be deep enough to test the subtrie
+	if !isTrieDeepEnough(activeTrie) {
+		// if not deep enough, keep re-trying
+		return createMultipleRandomTriesMini(t)
+	}
+
```

Comment:
This code checks if the last trie is deep enough, but other tries may still not be deep enough.  

Maybe we can check the first set of created tries (actually only the first one is sufficient to check), because all subsequent updated tries would be deep enough.

E.g. something like this to move `isTrieDeepEnough` to the block of code creating first set of tries:

```Go
func createMultipleRandomTriesMini(t *testing.T) []*trie.MTrie {
	tries := make([]*trie.MTrie, 0)
	activeTrie := trie.NewEmptyMTrie()

	var err error
	// add tries with no shared paths
        for len(tries) < 5 {
		paths, payloads := randNPathPayloads(20)
		activeTrie, _, err = trie.NewTrieWithUpdatedRegisters(activeTrie, paths, payloads, false)
		require.NoError(t, err, "update registers")		

		// First trie must be deep enough to test the subtries because subsequent
		// tries will be deep enough. To simplify test code, we can check all tries
		// instead of just the first trie.
		if isTrieDeepEnough(activeTrie) {
			tries = append(tries, activeTrie)
		}			

		// if not deep enough, keep re-trying
	}

	// add trie with some shared path
	sharedPaths, payloads1 := randNPathPayloads(10)
	activeTrie, _, err = trie.NewTrieWithUpdatedRegisters(activeTrie, sharedPaths, payloads1, false)
	require.NoError(t, err, "update registers")
	tries = append(tries, activeTrie)

	_, payloads2 := randNPathPayloads(10)
	activeTrie, _, err = trie.NewTrieWithUpdatedRegisters(activeTrie, sharedPaths, payloads2, false)
	require.NoError(t, err, "update registers")
	tries = append(tries, activeTrie)

	return tries
}
```

Reply:
Yeah, right. 

But I think that's OK, because we only care about the last trie, since the read leaf nodes only works for a single trie.

I updated so that the `createMultipleRandomTriesMini` returns the last trie, and the caller doesn't need to guess which trie is deep enough. 

I found this way the logic is simple.